java/registry/src/main/java/io/opensaber/registry/shard/advisory/SerialNumberShardAdvisor.java

```diff
@@ -0,0 +1,45 @@
+package io.opensaber.registry.shard.advisory;
+
+import org.springframework.stereotype.Component;
+
+import io.opensaber.registry.model.DBConnectionInfo;
+import io.opensaber.registry.model.DBConnectionInfoMgr;
+
+@Component
+public class SerialNumberShardAdvisor implements IShardAdvisor {
+
+	private String shardId;
+	private DBConnectionInfoMgr dBConnectionInfoMgr;
+
+	public SerialNumberShardAdvisor(DBConnectionInfoMgr dBConnectionInfoMgr) {
+		this.dBConnectionInfoMgr = dBConnectionInfoMgr;
+	}
+
+	@Override
+	public DBConnectionInfo getShard(Object serialNumber) {
+
+		DBConnectionInfo connectionInfo = null;
+		if (serialNumber != null) {
```

Comment:
I find this is too complex for an example. Why is a serialNumber not an integer?
if it is not because of schema definitions, lets correct it.
This must be a simple switch case, `slNum % 2 == 0` implies it is even and so shard X; otherwise shard Y.

Reply:
Yes right kept it for some default shards. Done now.